File: example_menu.py
"""Example of interactive menu."""
from kcmpy import App, Rect
from kcmpy.cui.menu import Menu, MenuItem

def on_new():
    print("\n✓ New project created")

def on_open():
    print("\n✓ Project opened")

def on_settings():
    print("\n✓ Settings opened")

def on_exit():
    print("\n✓ Exiting program")
    app.stop()

# Create menu (positioned below the text)
menu = Menu(
    Rect(5, 10, 40, 8),
    title="Main Menu",
    items=[
        MenuItem("New Project", on_new, "n"),
        MenuItem("Open Project", on_open, "o"),
        MenuItem("Settings", on_settings, "s"),
        MenuItem("Exit", on_exit, "q"),
    ]
)

print("=" * 50)
print("Interactive Menu - CUI Framework")
print("=" * 50)
print("\nControls:")
print("  ↑/↓  - navigate")
print("  Enter - select")
print("  n/o/s/q - hotkeys")
print()

# Run application
app = App(menu, inline=True)
try:
    app.run()
except KeyboardInterrupt:
    app.stop()

print("\nThank you for using KCM!")

File: kcmjs/example_menu.js
/**
 * Menu example for KCM Framework
 */
import { App, Rect, Menu, MenuItem, StatusIndicator, Component } from './main.js';

// Create root container
const root = new Component(new Rect(1, 1, 80, 25));

// Create status indicator
const status = new StatusIndicator(new Rect(2, 2, 60, 1), 'info', 'Используйте стрелки для навигации');
root.addChild(status);

// Create menu
const menu = new Menu(new Rect(20, 5, 40, 10), {
    title: 'Главное меню',
    items: [
        new MenuItem('Новый проект', () => {
            status.setStatus('success', 'Создан новый проект');
        }, 'n'),
        new MenuItem('Открыть проект', () => {
            status.setStatus('success', 'Открыт проект');
        }, 'o'),
        new MenuItem('Настройки', () => {
            status.setStatus('info', 'Открыты настройки');
        }, 's'),
        new MenuItem('Выход', () => app.stop(), 'q'),
    ]
});
root.addChild(menu);

// Create app
const app = new App(root, { inline: false, animated: false });
app.run();
